eth/contracts/DFDiamond.sol

```diff
@@ -0,0 +1,93 @@
+pragma solidity ^0.8.0;
+
+import {Ownable, OwnableInternal, OwnableStorage} from "@solidstate/contracts/access/ownable/Ownable.sol";
+import {IERC173} from "@solidstate/contracts/access/IERC173.sol";
+import {ERC165, IERC165, ERC165Storage} from "@solidstate/contracts/introspection/ERC165.sol";
+import {DiamondBase, DiamondBaseStorage} from "@solidstate/contracts/proxy/diamond/base/DiamondBase.sol";
+import {DiamondReadable, IDiamondReadable} from "@solidstate/contracts/proxy/diamond/readable/DiamondReadable.sol";
+import {DiamondWritable, IDiamondWritable} from "@solidstate/contracts/proxy/diamond/writable/DiamondWritable.sol";
+
+/**
+ * @title SolidState "Diamond" proxy reference implementation
+ */
+contract DFDiamond is DiamondBase, DiamondReadable, DiamondWritable, Ownable, ERC165 {
+    using DiamondBaseStorage for DiamondBaseStorage.Layout;
+    using ERC165Storage for ERC165Storage.Layout;
+    using OwnableStorage for OwnableStorage.Layout;
+
+    constructor() {
+        ERC165Storage.Layout storage erc165 = ERC165Storage.layout();
+        bytes4[] memory selectors = new bytes4[](10);
+
+        // register DiamondWritable
+
+        selectors[0] = IDiamondWritable.diamondCut.selector;
+
+        erc165.setSupportedInterface(type(IDiamondWritable).interfaceId, true);
+
+        // register DiamondReadable
+
+        selectors[1] = IDiamondReadable.facets.selector;
+        selectors[2] = IDiamondReadable.facetFunctionSelectors.selector;
+        selectors[3] = IDiamondReadable.facetAddresses.selector;
+        selectors[4] = IDiamondReadable.facetAddress.selector;
+
+        erc165.setSupportedInterface(type(IDiamondReadable).interfaceId, true);
+
+        // register ERC165
+
+        selectors[5] = IERC165.supportsInterface.selector;
+
+        erc165.setSupportedInterface(type(IERC165).interfaceId, true);
+
+        // register Ownable
+
+        selectors[6] = Ownable.owner.selector;
+        selectors[7] = Ownable.transferOwnership.selector;
+
+        erc165.setSupportedInterface(type(IERC173).interfaceId, true);
+
+        // register Diamond
+
+        selectors[8] = DFDiamond.getFallbackAddress.selector;
+        selectors[9] = DFDiamond.setFallbackAddress.selector;
+
+        // diamond cut
+
+        FacetCut[] memory facetCuts = new FacetCut[](1);
+
+        facetCuts[0] = FacetCut({
+            target: address(this),
+            action: IDiamondWritable.FacetCutAction.ADD,
+            selectors: selectors
+        });
+
+        DiamondBaseStorage.layout().diamondCut(facetCuts, address(0), "");
+
+        // set owner
+
+        OwnableStorage.layout().setOwner(msg.sender);
+    }
+
+    receive() external payable {}
+
+    /**
+     * @notice get the address of the fallback contract
+     * @return fallback address
+     */
+    function getFallbackAddress() external view returns (address) {
+        return DiamondBaseStorage.layout().fallbackAddress;
```

Comment:
Do we know if this is 0x0 by default?

Reply:
0x0000000000000000000000000000000000000000 from my test

Reply:
For my own reference: If a function is called on the diamond and a facet containing that signature isn't found, the Diamond will call delegatecall that function to the fallback address. 

If the fallback address is `0x00`, the call will revert with `'DiamondBase: no facet found for function signature'`.

I think we should never set a fallback address because we want to fail if a facet is never found.

Reply:
[eee9239](https://github.com/dfdao/game/pull/7/commits/eee92399f96f1cdacd41483c665bb1bf9f129761) just to be safe